src/compiler/parser.ts

```diff
@@ -3221,6 +3221,37 @@ module ts {
                     }
                 }
 
+                // If encounter "([", this could be the start of an array binding pattern.
+                // Examples:
+                //      ([ x ]) => { }
+                //      ([ x ])
+                if (second === SyntaxKind.OpenBracketToken) {

```

Comment:
But won't your new code return ' unknown' on this case?  I'm curious what cases were Will see that are common where we'll be able to effectively bail out early with this code b

    -- Cyrus
________________________________
From: Ron Buckton<mailto:notifications@github.com>
Sent: ‎4/‎10/‎2015 9:20 AM
To: Microsoft/TypeScript<mailto:TypeScript@noreply.github.com>
Cc: Cyrus Najmabadi<mailto:cyrusn@microsoft.com>
Subject: Re: [TypeScript] Add support for parsing binding patterns in arrow function parameters (#2680)


In src/compiler/parser.ts<https://github.com/Microsoft/TypeScript/pull/2680#discussion_r28158593>:

> @@ -3221,6 +3221,37 @@ module ts {
>                      }
>                  }
>
> +                // If encounter "([", this could be the start of an array binding pattern.
> +                // Examples:
> +                //      ([ x ]) => { }
> +                //      ([ x ])
> +                if (second === SyntaxKind.OpenBracketToken) {


I'm seeing it often in conjunction with Promise.all:

Promise.all([a, b, c]).then(([a, b, c]) => {
});

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/2680/files#r28158593>.